### PR TITLE
Common: add generic checker for trending plots

### PR DESCRIPTION
**Pull request by aferrero2707**
The checker is designed to verify that the values of trend graphs are within given minimum and maximum limits.
The limits can be specified as fixed values, or values relative to either the mean or the standard deviation of a given set of graph points.

See https://its.cern.ch/jira/browse/QC-1216 for more details.

### CODE REVIEW

**Comment by aferrero2707** [comment]
@Barthelemy @knopers8 this PR is rather big, but it does not touch existing code, it only introduces a new checker + some utility class to handle the configuration parameters.
Once it will be possible to retrieve the current interaction rate from CTP, the code is already designed to take advantage of it by specifying rate-dependent checker thresholds in the configuration.

**Comment by aferrero2707** [comment]
@Barthelemy would you have the possibility to review this PR in the next days? Apart from the support for rate-dependent checker thresholds (which might only be useful at a later stage), the code could be already used to check the trends of the raw and CTF sizes, for example, and to replace the custom checker for the vertex positions and sigmas.
Thanks!

**Comment by Barthelemy** [comment]
@aferrero2707 yes, apologies, this slipped through some hole and disappeared from my radar.

File: Modules/Common/src/TrendCheck.cxx
Comment (Barthelemy) [inline Modules/Common/src/TrendCheck.cxx:152]: lower case and if possible descriptive name

**Review by Barthelemy — COMMENTED** [review]
Very good, thanks. 

Could you perhaps add a bit of doc ? 
I am thinking that we should split the doc further as it becomes very large. We could have a separate page for the "commons". 
I'll do it once this is merged.

**Comment by aferrero2707** [comment]
@Barthelemy thanks for reviewing!

> Could you perhaps add a bit of doc ? 

Yes, I was indeed planning to do that, I just wanted to wait for the first feedback in case it would required significant re-work.
What it the appropriate documentation file for putting the documentation for checkers?

**Comment by Barthelemy** [comment]
@aferrero2707 You can put it in Advanced.md for now. I will refactor the doc and move all the info about common in a separate file.

File: Modules/Common/src/TrendCheck.cxx
Comment (aferrero2707) [reply Modules/Common/src/TrendCheck.cxx]: Done.

**Comment by aferrero2707** [comment]
@Barthelemy I did the suggested change and added the documentation.

**Review by Barthelemy — APPROVED** [review]
Great ! thank you

**Pull request merged**
